pywt/_extensions/_pywt.pxd

```diff
@@ -8,7 +8,8 @@ ctypedef fused data_t:
     np.float64_t
 
 cdef public class Wavelet [type WaveletType, object WaveletObject]:
-    cdef wavelet.Wavelet* w
+    cdef wavelet.DiscreteWavelet* dw
+    cdef wavelet.ContinuousWavelet* cw
```

Comment:
If we split `Wavelet` into `ContinuousWavelet` and `DiscreteWavelet`, then this would be much tidier. The `if dw != NULL` checks below would also all fall away, since those properties would only make sense on the `DiscreteWavelet`.